### ISSUE TITLE
Create the DataStore::getAllExistingKeys() method

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/ddpdemo#24]
and add it to the HDF5DataStore class.

As we discussed, this method should return a list of StorageKeys, one key per fragment that is found in the DataStore.

In HDF5DataStore, this method will need to be sophisticated enough to understand how to handle different fragment groups (all groups in one file, fragments grouped into event-based files, and one fragment per file)

**Comment by bieryAtFnal** [comment]
Something that I realized when I started to work on this...
It would be helpful in the demo.hdf5 file from Sprint 1 if the event-ID-based top-level groups were zero-padded.  The h5dump and HDFView utilities (and maybe HDF5 itself) sorts strings lexically, and I believe that people will find it easier to understand the data if "11" comes after "2" rather than before.
If we agree that we would like to zero-pad the EventID (e.g. "0001"), then I will create a dedicated Issue for that.

**Comment by bieryAtFnal** [comment]
Another observation: we may want/need to change the 

`new HighFive::File(path + fileName, HighFive::File::OpenOrCreate | HighFive::File::Truncate);`

call to 

'new HighFive::File(path + fileName, HighFive::File::OpenOrCreate);`

so that contents of the file don't get destroyed when we open it for reading.

**Comment by bieryAtFnal** [comment]
I withdraw my earlier comment on the flags that we pass to the HighFive::File constructor: the HighFile::File creation will likely move into other methods than the constructor, and I expect that the open/create flags will be fairly obvious once we do that.

**Comment by bieryAtFnal** [comment]
I have created a branch called `feature/Issue24_getAllKeys` and pushed some code changes on that branch to the central repo.  The code changes are far from complete, but they are test-able, so I wanted to make this snapshot available centrally.

On this branch, there is now a `GetAllKeysTest` class in the `src` directory and a `get_all_keys_test.json` config file in the `test` directory.  I've used these to test the addition of the initial `getAllExistingKeys()` method in HDF5DataStore.

To do that test, I used the following steps:

- ran our old DiskWriterDemo to generate a data file named `demo.hdf5` [let's say that I generated N events in that run]
- renamed that file to `demo_event1.hdf5` (Yes, I know that this file has data from many events, but I'm pretending that it has multiple fragments from a single event.)
- ran our old DiskWriterDemo **again** to generate a second data file named `demo.hdf5` [let's say that I generated M events in that run]
- renamed the second `demo.hdf5` file as `demo_event2.hdf5`
- ran the get_all_keys test using this command:  `daq_application -c QueryResponseCommandFacility -j sourcecode/ddpdemo/test/get_all_keys_test.json`
- verified that the ERS_INFO statement that gets printed out by the GetAllKeysDemo says that it found `N+M` Keys

In this code so far, I've assumed  that our filename patterns are strings like `demo.hdf5`, `demo_event%e.hdf5`,  and `demo_event%e_geo%g.hdf5`.  The `get_all_keys_test.json` config file currently specifies a filename pattern of `demo_event%e.hdf5` and that is why I chose the names above when I renamed the generated `demo.hdf5` files.

Within the HDF5DataStore class, I created a getAllFiles_() private method to take the specified path and filename pattern to look up all files that match the pattern.

**Comment by bieryAtFnal** [comment]
I've merged the `develop` branch into the `feature/Issue24_getAllKeys` branch (after Pull Request #35 (KeyTranslator) was merged to `develop`.  And, I updated the HDF5DataStore::getAllExistingKeys() method to make use of one of the KeyTranslator methods.  The merged code looks (and runs) fine.

I had been thinking that I would submit a merge request for the getAllExistingKeys() work next, but I noticed that I've change a couple of things in the HDF5DataStore constructor that might conflict with the work that Adam is doing (e.g. I changed the filename constructor argument to a filePattern).  So, maybe I should wait.

it would be great to talk in our daily meeting today about which PR should go first.  Probably the most important inputs to that choice are A) how close Adam is to his next PR  and B) how urgently Carlos would like to start using getAllExistingKeys().
